gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineImmutables.java

```diff
@@ -16,23 +16,33 @@
 
 package com.palantir.baseline.plugins;
 
+import java.util.Collections;
 import java.util.Objects;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
 import org.gradle.api.artifacts.Dependency;
-import org.gradle.api.plugins.JavaPluginConvention;
 import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.SourceSetContainer;
 import org.gradle.api.tasks.compile.JavaCompile;
 
 public final class BaselineImmutables implements Plugin<Project> {
 
     @Override
     public void apply(Project project) {
         project.getPluginManager().withPlugin("java", unused -> {
-            project.afterEvaluate(proj -> {
-                proj.getConvention().getPlugin(JavaPluginConvention.class).getSourceSets().stream()
```

Comment:
Calling `.stream()` on a `DomainObjectCollection` is pretty much always a bug, as it doesn't include objects added later (and encourages people to use afterEvaluate). I wonder if we should make this an error prone check?

Reply:
Sounds like a great idea

Reply:
That said, we do use a `.stream()` below correctly....